app/src/main/java/edu/outside2154/gamesense/fragment/SettingsFragment.kt

```diff
@@ -41,69 +56,79 @@ class SettingsFragment : Fragment() {
+
+        btnSubmit.setOnClickListener {
+            atkGoal = atkHours.text.toString().trim()
+            intGoal = intHours.text.toString().trim()
+            hthGoal = hthHours.text.toString().trim()
+            if(atkGoal.length > 0) {
+                updateAtk = true
+            }
+            if(intGoal.length > 0) {
+                updateInt = true
+            }
+            if(hthGoal.length > 0) {
+                updateHth = true
+            }
+            if (atkGoal.length > 0 || intGoal.length > 0 || hthGoal.length > 0) {
+                update()
+                Toast.makeText(getActivity(), "Goals have been updated!", Toast.LENGTH_SHORT).show()
+            } else {
+                Toast.makeText(getActivity(), "Please select all options!", Toast.LENGTH_SHORT).show()
+            }
+        }
         return v
-
     }
 
-    // TODO: Rename method, update argument and hook method into UI event
-    fun onButtonPressed(uri: Uri) = mListener?.onFragmentInteraction(uri)
-
-    override fun onDetach() {
-        super.onDetach()
-        mListener = null
-    }
-
-    /**
-     * This interface must be implemented by activities that contain this
-     * fragment to allow an interaction in this fragment to be communicated
-     * to the activity and potentially other fragments contained in that
-     * activity.
-     *
-     *
-     * See the Android Training lesson [Communicating with Other Fragments](http://developer.android.com/training/basics/fragments/communicating.html) for more information.
-     */
-    interface OnFragmentInteractionListener {
-        // TODO: Update argument type and name
-        fun onFragmentInteraction(uri: Uri)
-    }
-
-    companion object {
-        // TODO: Rename parameter arguments, choose names that match
-        // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
-        private val ARG_PARAM1 = "param1"
-        private val ARG_PARAM2 = "param2"
-
-        /**
-         * Use this factory method to create a new instance of
-         * this fragment using the provided parameters.
-         *
-         * @param param1 Parameter 1.
-         * @param param2 Parameter 2.
-         * @return A new instance of fragment SettingsFragment.
-         */
-        // TODO: Rename and change types and number of parameters
-        fun newInstance(param1: String, param2: String): SettingsFragment {
-            val fragment = SettingsFragment()
-            fragment.arguments = Bundle().apply {
-                putString(ARG_PARAM1, param1)
-                putString(ARG_PARAM2, param2)
+    override fun update() {
+            val androidId = getAndroidId(activity)
+
+            if(updateAtk == true) {
+                FirebaseDatabase.getInstance().reference.child("$androidId/atkStat/current/").removeValue()
+                FirebaseDatabase.getInstance().reference.child("$androidId/atkStat/goals/").removeValue()
```

Comment:
Can we add a comment describing what's happening here? It looks like we are clearing out the current stat and replacing it with a map with one entry. Is that correct?

This also could lead to a bug if we set the same goal as we had before - our progress for that goal will be gone.